tests/test_agents/base.py

```diff
@@ -465,22 +465,27 @@ def _stop_and_restart_last_agent(self, nb_agents: int) -> None:
             timeout=self.wait_before_stop,
         )
         if missing_strict_strings:
-            raise RuntimeError("cannot stop agent correctly")
-        logging.info("Last agent stopped")
+            msg = f"cannot stop agent, stop string `{self.stop_string}` not found"
+            raise RuntimeError(msg)
+
+        if self.wait_to_kill:
+            logging.info("Waiting to terminate agents")
+            time.sleep(self.wait_to_kill)
 
-        # immediately terminate the agent
-        self.terminate_agents(self.processes[-1], timeout=0)
+        # immediately terminate the agents - sequentially
         # don't pop before termination, seems to lead to failure!
-        self.processes.pop()
+        for _ in range(self.n_terminal):
+            self.terminate_agents(self.processes[-1], timeout=0)
+            self.processes.pop()
+            logging.info("Last agent stopped")
```

Comment:
We may want to stop popping the last agent. It only adds unnecessary complexity. We could instead:

```suggestion
        for i in range(self.n_terminal):
            self.terminate_agents(self.processes[i], timeout=0)
            self.processes.pop(0)
            logging.info(f"Agent {i} stopped")
```

Reply:
I agree with that, but I did not want too change too much at once in fear or breaking things and slowing down